src/results.cpp

```diff
@@ -370,20 +374,24 @@ util::Optional<Mixed> Results::aggregate(size_t column, bool return_none_for_emp
 
 util::Optional<Mixed> Results::max(size_t column)
 {
-    return aggregate(column, true, "max",
-                     [=](auto const& table) { return table.maximum_int(column); },
-                     [=](auto const& table) { return table.maximum_float(column); },
-                     [=](auto const& table) { return table.maximum_double(column); },
-                     [=](auto const& table) { return table.maximum_timestamp(column); });
+    size_t return_ndx;
+    auto results = aggregate(column, true, "max",
+                     [&](auto const& table) { return table.maximum_int(column, &return_ndx); },
+                     [&](auto const& table) { return table.maximum_float(column, &return_ndx); },
+                     [&](auto const& table) { return table.maximum_double(column, &return_ndx); },
+                     [&](auto const& table) { return table.maximum_timestamp(column, &return_ndx); });
+    return return_ndx == npos ? none : results;
```

Comment:
Good point, and the only exception is the average. I will just check table empty there.